### DISCUSSION TITLE
The `Move right` cursor become so slow in `V-BLOCk` after mapping the `l` key to `i`

### DISCUSSION THREAD

**Discussion by azzamsa** [LazyVim/LazyVim#2220] [Q&A] [answered]
Hi.

Inside `V-BLOCK`, the cursor becomes very slow compared to using the `arrow keys` when I hit `i` repeatedly (remapped from `l`).

As a temporary workaround, I use the arrow keys inside `V-BLOCK`. 
Any suggestions are very appreciated. Using `hjkl` as the default movement is currently not an option.

```lua
local map = vim.keymap.set

-- Insert Key
map({ "n", "x" }, "u", "i", { desc = "" })
map({ "n", "x" }, "U", "I", { desc = "" })

-- Undo `operations`
--
map({ "n", "x" }, "l", "u", { desc = "" })
map({ "n", "x" }, "L", "U", { desc = "" })

-- Cursor Movement
--     ^
--     e
-- < m   i >
--     n
--     v

map({ "n", "x" }, "m", "h", { desc = "" })
map({ "n", "x" }, "M", "H", { desc = "" })

map({ "n", "x" }, "n", "j", { desc = "" })
map({ "n", "x" }, "N", "J", { desc = "" })

map({ "n", "x" }, "e", "k", { desc = "" })
map({ "n", "x" }, "E", "K", { desc = "" })

map({ "n", "x" }, "i", "l", { desc = "" })
map({ "n", "x" }, "I", "L", { desc = "" })

map({ "n", "x" }, "n", "v:count == 0 ? 'gj' : 'j'", { expr = true, silent = true })
map({ "n", "x" }, "<Down>", "v:count == 0 ? 'gj' : 'j'", { expr = true, silent = true })

map({ "n", "x" }, "e", "v:count == 0 ? 'gk' : 'k'", { expr = true, silent = true })
map({ "n", "x" }, "<Up>", "v:count == 0 ? 'gk' : 'k'", { expr = true, silent = true })
```

Full config: https://github.com/azzamsa/roof/blob/master/lua/config/keymaps.lua

**Comment by dpetka2001** [comment, marked as answer]
If you do `:verbose xmap i` it will show you that it is mapped to some text objects that come from [mini.indentscope](https://www.lazyvim.org/plugins/ui#miniindentscope) and [mini.ai](https://www.lazyvim.org/plugins/coding#miniai). If you disable these 2 plugins you shouldn't observe the delay any more. I'm not sure if you can change the internal mappings of the plugins, you would have to read the docs of the plugins to do that.
